core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -550,19 +552,52 @@ public JSONArray getPlatformPluginUpdates() {
         updateSiteList: for (UpdateSite updateSite : Jenkins.get().getUpdateCenter().getSiteList()) {
             String updateCenterJsonUrl = updateSite.getUrl();
             String suggestedPluginUrl = updateCenterJsonUrl.replace("/update-center.json", "/platform-plugins.json");
+            VersionNumber version = Jenkins.getVersion();
+            if (version != null && (suggestedPluginUrl.startsWith("https://") || suggestedPluginUrl.startsWith("http://"))) {
+                // Allow remote update site to distinguish based on the current version
+                // This looks a bit hacky but UpdateCenter#toUpdateCenterCheckUrl does something similar
+                suggestedPluginUrl = suggestedPluginUrl + (suggestedPluginUrl.contains("?") ? "&" : "?") + "version=" + version;
+            }
             try {
                 URLConnection connection = ProxyConfiguration.open(new URL(suggestedPluginUrl));
                 
                 try {
+                    String initialPluginJson = IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
+
+                    JSONObject initialPluginObject = null;
+
                     if(connection instanceof HttpURLConnection) {
                         int responseCode = ((HttpURLConnection)connection).getResponseCode();
                         if(HttpURLConnection.HTTP_OK != responseCode) {
                             throw new HttpRetryException("Invalid response code (" + responseCode + ") from URL: " + suggestedPluginUrl, responseCode);
                         }
+
+                        if (DownloadService.signatureCheck) {
```

Comment:
Should it have a separate flag for update centers which do not sign the platform services file but sign other artifacts?
In such case we may end up in regressions

Reply:
No. I don't think anyone uses this undocumented feature for remote update sites, so it would be trivial to add if they start publishing this.